src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNames.cs

```diff
@@ -0,0 +1,292 @@
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+        internal static DiagnosticDescriptor UnreferencedParameterRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageUnreferencedParameter,
+                                                                             DiagnosticCategory.Design,
+                                                                             RuleLevel.BuildWarning,
+                                                                             description: s_localizableDescriptionUnreferencedParameter,
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(PropertyNameRule, FieldRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction((compilationStartContext) =>
+            {
+                INamedTypeSymbol? jsonConstructorAttributeNamedSymbol = compilationStartContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextJsonSerializationJsonConstructorAttribute);
+                if (jsonConstructorAttributeNamedSymbol == null)
+                {
+                    return;
+                }
+
+                var paramAnalyzer = new ParameterAnalyzer(jsonConstructorAttributeNamedSymbol);
+
+                compilationStartContext.RegisterSymbolStartAction((symbolStartContext) =>
+                {
+                    var constructors = ((INamedTypeSymbol)symbolStartContext.Symbol).InstanceConstructors;
+
+                    foreach (var ctor in constructors)
+                    {
+                        if (paramAnalyzer.ShouldAnalyzeMethod(ctor))
+                        {
+                            var referencedParameters = PooledConcurrentSet<IParameterSymbol>.GetInstance();
+
+                            symbolStartContext.RegisterOperationAction(
+                                context => ParameterAnalyzer.AnalyzeOperationAndReport(context, referencedParameters),
+                                OperationKind.ParameterReference);
+
+                            symbolStartContext.RegisterSymbolEndAction(
+                                context => ParameterAnalyzer.ReportUnusedParameters(context, ctor, referencedParameters));
```

Comment:
By accounting that the JsonSerializer will not throw for `unused parameters` I don't think we should warn for `unused parameters`, if we reduce the  analysis level to Info it will become just the same as `ReviewUnusedParametersAnalyzer` so i think we better not analyze this (plus it is causing to use shared concurrent set would affect perf)

Reply:
The exception will only be thrown on deserialization:

```
Each parameter in constructor 'Void .ctor(xxx)' on type 'YYY' must bind to an object property or field on deserialization. Each parameter name must match with a property or field on the object. The match can be case-insensitive.
```

During the serialization, there won't be any exceptions, but a value won't be assigned. So, if we care about deserialization too, then it may be necessary. It was my original motivation behind adding this logic.

I've removed it, but it can be easily reverted if needed.